apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ManagedTransaction.cs

```diff
@@ -0,0 +1,752 @@
+
+        private SpannerClient Client => _multiplexSession.Client;
+
+        /// <summary>
+        /// Whether the transaction associated to this session is single use or not.
+        /// If this is true then <see cref="TransactionOptions"/> will be <see cref="TransactionOptions.ModeOneofCase.ReadOnly"/>
+        /// and <see cref="TransactionId"/> will be null.
+        /// </summary>
+        internal bool SingleUseTransaction { get; }
+
+        /// <summary>
+        /// The ID of the transaction. May be null.
+        /// </summary>
+        /// <remarks>
+        /// Transactions are acquired when they are needed so this will be null in the
+        /// following cases:
+        /// <list type="bullet">
+        /// <item>
+        /// <see cref="TransactionOptions"/> is <see cref="TransactionOptions.ModeOneofCase.None"/>,
+        /// including when the session is idle in the pool.
+        /// </item>
+        /// <item>
+        /// <see cref="SingleUseTransaction"/> is true. No transactions will exist client side.
+        /// </item>
+        /// <item>
+        /// The session has been acquired from the pool with some transaction options but no command execution has
+        /// been attempted since.
+        /// </item>
+        /// <item>
+        /// The first command execution is underway and the transaction is being created either
+        /// by inlining or explicitly.
+        /// </item>
+        /// </list>
+        /// </remarks>
+        public ByteString TransactionId => Interlocked.CompareExchange(ref _transaction, null, null)?.Id;
+
+        /// <summary>
+        /// The read timestamp of the transaction. May be null.
+        /// </summary>
+        /// <remarks>
+        /// Will be set iif a transaction has been started and <see cref="TransactionOptions"/> represents read-only options
+        /// with <see cref="Types.ReadOnly.ReturnReadTimestamp"/> set to true
+        /// or if this session was created using <see cref="SessionPool.CreateDetachedSession(SessionName, ByteString, ModeOneofCase, Timestamp)"/>
+        /// and a value was provided for the <see cref="Timestamp"/> parameter.
+        /// </remarks>
+        public Timestamp ReadTimestamp => Interlocked.CompareExchange(ref _transaction, null, null)?.ReadTimestamp;
+
+        // internal for testing
+        internal MultiplexedSessionPrecommitToken PrecommitToken { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="multiplexSession"></param>
+        /// <param name="transactionId"></param>
+        /// <param name="transactionOptions"></param>
+        /// <param name="singleUseTransaction"></param>
+        /// <param name="readTimestamp"></param>
+        public ManagedTransaction(ManagedSession multiplexSession, ByteString transactionId, TransactionOptions transactionOptions, bool singleUseTransaction, Timestamp readTimestamp)
```

Comment:
Private constructors and factory methods, this is to protect us from future breaking changes.